.github/workflows/build-beta.yml

```diff
@@ -0,0 +1,139 @@
+---
+name: build-beta
+
+on:
+  push:
+    branches:
+      # TODO: remove the `gha-` to match the current workflow again
+      - build-gha-\d+$
+      - build-gha-ios-\d+$
+      - build-gha-ios-beta-\d+$
+      - build-gha-ios-sim-\d+$
+      - build-gha-android-\d+$
+      - build-gha-android-beta-\d+$
+
+env:
+  NODE_VERSION: 18.7.0
+  TERM: xterm
+
+jobs:
+  test:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: ci/checkout-repo
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: ci/test
+        uses: ./.github/actions/test
+
+  build-and-deploy-android-beta:
+    runs-on: ubuntu-22.04
+    if: ${{ !startsWith(github.ref_name , 'build-ios') }}
+    needs:
+      - test
+    steps:
+      - name: ci/checkout-repo
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+
+      - name: ci/prepare-android-build
+        uses: ./.github/actios/prepare-android-build
+
+      - name: ci/build-and-deploy-android-beta
+        env:
+          AWS_ACCESS_KEY_ID: "${{ secrets.MM_MOBILE_BETA_AWS_ACCESS_KEY_ID }}"
+          AWS_SECRET_ACCESS_KEY: "${{ secrets.MM_MOBILE_BETA_AWS_SECRET_ACCESS_KEY }}"
+          MATTERMOST_WEBHOOK_URL: "${{ secrets.MM_MOBILE_BETA_MATTERMOST_WEBHOOK_URL }}"
+          SENTRY_AUTH_TOKEN: "${{ secrets.MM_MOBILE_SENTRY_AUTH_TOKEN }}"
+          SENTRY_DSN_ANDROID: ${{ secrets.MM_MOBILE_BETA_SENTRY_DSN_ANDROID }}
+          SUPPLY_JSON_KEY: ${{ github.workspace }}/mattermost-mobile-private/android/mattermost-credentials.json
+        run: |
+          echo "::group::Build"
+          bundle exec fastlane android build --env android.beta
+          echo "::endgroup::"
+          echo "::group::Deploy to Play Store"
+          bundle exec fastlane android deploy file:"${{ github.workspace }}/*.apk" --env android.beta
+          echo "::endgroup::"
+        working-directory: ./fastlane
+
+      - name: ci/upload-android-beta-build
+        uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3.1.2
+        with:
+          name: android-build-beta-${{ github.run_id }}
+          path: "*.apk"
+
+  build-ios-simulator:
+    runs-on: macos-12
+    if: ${{ !startsWith(github.ref_name , 'build-android') }}
+    needs:
+      - test
+    steps:
+      - name: ci/checkout-repo
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+
+      - name: ci/prepare-ios-build
+        uses: ./.github/actios/prepare-ios-build
+
+      - name: ci/build-ios-simulator
+        env:
+          TAG: "${{ github.ref_name }}"
+          MATTERMOST_WEBHOOK_URL: "${{ secrets.MM_MOBILE_BETA_MATTERMOST_WEBHOOK_URL }}"
```

Comment:
thanks for adding this, and sorry but I just remembered this also needs the AWS env vars

Reply:
@enahum the context used for the unsigned build on CircleCI was `mattermost-mobile-unsigned` . 
It only had the below secrets declared. 
```
Using environment variables from project settings and/or contexts:
  CIRCLE_OIDC_TOKEN=**REDACTED**
  CIRCLE_OIDC_TOKEN_V2=**REDACTED**
  GITHUB_TOKEN=**REDACTED**
  MATTERMOST_WEBHOOK_URL=**REDACTED**
  NODE_OPTIONS=**REDACTED**
  TAG=**REDACTED**
```

Are you sure AWS vars are needed ?

Reply:
Basically that is needed so that the simulator build gets uploaded to aws and a message posted using the incoming webhook (see Fastfile) and that is something that the QA team is asking for (cc: @saturninoabril )

Btw in the meantime I'll also add them to the circleCI context

Reply:
Cool . Will do . Do you know the rest of the AWS env vars needed which are not secrets ? 
```
AWS_BUCKET_NAME=
AWS_FOLDER_NAME=
AWS_REGION=us-east-1
```

Reply:
Thanks for adding!